superset/security.py

```diff
@@ -147,23 +151,29 @@ def datasource_access_by_name(
                 return True
         return False
 
-    def datasource_access_by_fullname(
-            self, database, full_table_name, schema):
-        table_name_pieces = full_table_name.split('.')
+    def get_schema_and_table(self, table_in_query, schema):
+        table_name_pieces = table_in_query.split('.')
         if len(table_name_pieces) == 2:
             table_schema = table_name_pieces[0]
             table_name = table_name_pieces[1]
         else:
             table_schema = schema
             table_name = table_name_pieces[0]
+        return (table_schema, table_name)
+
+    def datasource_access_by_fullname(
+            self, database, table_in_query, schema):
+        table_schema, table_name = self.get_schema_and_table(table_in_query, schema)
         return self.datasource_access_by_name(
             database, table_name, schema=table_schema)
 
     def rejected_datasources(self, sql, database, schema):
         superset_query = sql_parse.SupersetQuery(sql)
-        return [
+        rejected = [
             t for t in superset_query.tables if not
             self.datasource_access_by_fullname(database, t, schema)]
+        if rejected:
+            return [':'.join(self.get_schema_and_table(i, schema)) for i in rejected]
```

Comment:
This should be removed (discussed offline). Returning a list of tables (w/ schema) seems suffice.

Reply:
Agree. Done.